src/v/storage/BUILD

```diff
@@ -82,24 +82,25 @@ redpanda_cc_library(
     hdrs = [
         "index_state.h",
     ],
-    include_prefix = "storage",
-    deps = [
-        ":logger",
+    implementation_deps = [
         "//src/v/base",
-        "//src/v/bytes:iobuf",
         "//src/v/bytes:iobuf_parser",
-        "//src/v/container:fragmented_vector",
-        "//src/v/features",
         "//src/v/hashing:crc32c",
         "//src/v/hashing:xx",
-        "//src/v/model",
         "//src/v/reflection:adl",
-        "//src/v/serde",
         "//src/v/serde:bool_class",
+        "//src/v/serde:iobuf",
         "//src/v/serde:optional",
         "//src/v/serde:vector",
         "//src/v/utils:to_string",
         "@fmt",
+    ],
+    include_prefix = "storage",
+    deps = [
+        "//src/v/bytes:iobuf",
+        "//src/v/container:fragmented_vector",
+        "//src/v/model",
+        "//src/v/serde",
```

Comment:
What does it mean for //src/v/serde to be in deps but //src/v/serde:bool_class, etc to be in implementation_deps?

Reply:
serde isn't a superset, it just provides the basics like envelope. serde::bool_class is what enables support for serializing/deserializing bool_class type.

Reply:
Makes sense, thanks!